Makefile

```diff
@@ -67,11 +64,7 @@ $(BIN).exe: $(SRC) | $(VAPIS)
 
 $(BIN): $(SRC) | $(VAPIS)
 	@echo 'Compiling $(VALA_FILTER) -> $@'
-ifeq ($(VALAPKG),libvala-0.40)
-	$(VALAC) --vapidir=/usr/local/share/vala/vapi -o $@ --pkg posix $(VALA040) --save-temps ${TEMPS}
-else
-	$(VALAC) -o $@ --pkg posix --pkg $(VALAPKG) --pkg valacodegen-0.40 --save-temps ${TEMPS}
-endif
+	$(VALAC) -o $@ --pkg posix $(VALA_PRIVATE_CODEGEN) --save-temps ${TEMPS}
```

Comment:
that will break in old valas

Reply:
Using vala < 0.40 was already broken before afaics
Obviously this change will work with vala 0.38.5+ only, so additional conditions are required